File: app/src/main/java/com/example/schoolairdroprefactoredition/scene/user/ResultViewModel.java
package com.example.schoolairdroprefactoredition.scene.user;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.example.schoolairdroprefactoredition.presenter.callback.IResultCallback;
import com.example.schoolairdroprefactoredition.presenter.impl.UserNameImpl;
import com.example.schoolairdroprefactoredition.scene.main.base.BaseStateViewModel;

public class ResultViewModel extends BaseStateViewModel implements IResultCallback {

    private MutableLiveData<Boolean> mRenameResult = new MutableLiveData<>();

    private UserNameImpl userNameImpl;

    public ResultViewModel() {
        userNameImpl = new UserNameImpl();
        userNameImpl.registerCallback(this);
    }

    public LiveData<Boolean> rename(String token, String name) {
        userNameImpl.rename(token, name);
        return mRenameResult;
    }

    @Override
    public void onSuccess() {
        mRenameResult.postValue(true);
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        userNameImpl.unregisterCallback(this);
    }

    @Override
    public void onError() {
        mOnRequestListener.onError();
    }

}

File: app/src/main/java/com/example/schoolairdroprefactoredition/scene/user/ResultViewModel.kt
package com.example.schoolairdroprefactoredition.scene.user

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.schoolairdroprefactoredition.repository.UserNameRepository
import kotlinx.coroutines.launch

class ResultViewModel : ViewModel() {

    private val mRenameResult = MutableLiveData<Boolean>()

    private val userNameRepository by lazy {
        UserNameRepository.getInstance()
    }

    fun rename(token: String, name: String): LiveData<Boolean> {
        viewModelScope.launch {
            userNameRepository.rename(token, name) {
                mRenameResult.postValue(it)
            }
        }
        return mRenameResult
    }
}